tests/component/mock_gw/gw_backend.py

```diff
@@ -0,0 +1,357 @@
+        """
+        return err, result
+        sqlglot may be useful
+        """
+        sql = sql.strip().lower()
+
+        # case set k=v, simple ignore
+        if sql.startswith("set"):
+            return "OK", "set"
+
+        # case: use catalog xx_cat
+        # case: use xx_db
+        if sql.startswith("use"):
+            if group := re.search(re_use_cat_db, sql):
+                return self._use_cat_db(group[1], group[2]), "use_catalog_db"
+            if group := re.search(re_use_cat, sql):
+                return self._use_cat_db(group[1], None), "use_catalog"
+            if group := re.search(re_use_db, sql):
+                return self._use_cat_db(None, group[1]), "use_database"
+            raise RuntimeError(f"unsupported sql {sql}")
+
+        # case: show catalogs
+        # case: show databases
+        # case: show current catalog
+        # case: show current database
+        elif sql.startswith("show"):
+            if re.search(re_show_cats, sql):
+                return self._show_catalogs(), "show_catalogs"
+            elif re.search(re_show_dbs, sql):
+                return self._show_databases(), "show_databases"
+            elif re.search(re_show_curr_cat, sql):
+                return self._show_current_catalog(), "show_current_catalog"
+            elif re.search(re_show_curr_db, sql):
+                return self._show_current_database(), "show_current_database"
+            elif match := re.search(re_show_tbls, sql):
+                catalog = match[3][:-1] if match[3] else None
+                database = match[5] if match[5] else None
+                return self._show_tables(catalog, database), "show_tables"
+            elif match := re.search(re_show_views, sql):
+                catalog = match[3][:-1] if match[3] else None
+                database = match[5] if match[5] else None
+                return self._show_views(catalog, database), "show_views"
+            else:
+                raise RuntimeError(f"unsupported sql {sql}")
+
+        # create database xx_db
+        elif match := re.search(re_create_db, sql):
+            db_name = match.group(1)
+            return self._db_create(db_name), "create_database"
+
+        # execute by sqlite
+        parsed_sql = sqlglot.parse_one(sql)
+        if parsed_sql.key == "create":
+            is_table = GwBackend._is_table_not_view_when_create_or_drop(sql)
+            return self._table_create(sql, is_table), "create_table" if is_table else "create_view"
+        elif parsed_sql.key == "select":
+            return self._table_select(sql), "_type"
+        elif parsed_sql.key == "insert":
+            return self._table_insert(sql), "_type"
+        elif parsed_sql.key == "drop":
+            is_table = GwBackend._is_table_not_view_when_create_or_drop(sql)
+            return self._table_drop(sql, is_table), "_type"
+        else:
+            raise "NOT support"
+
+    def _execute_by_sqlite(self, sql) -> list[Any]:
```

Comment:
Return type should be `List[Any]` imported from `typing`

Reply:
done